### ISSUE TITLE
When clipman item store runs full, clipman discards the current copy buffer.

### ISSUE THREAD

**Issue by bionade24** [yory8/clipman#90]
I delayed this behaviour a bit by setting `--max-items=1000`.   But until I restart the clipman session, it just discards any copy buffer. Is this intended or am I doing something wrong?

**Comment by yory8** [comment]
Do you mean that it stops adding new items? That's not what should happen, the expected behavior is that it drops the oldest item in memory to make space for the new ones.

**Comment by bionade24** [comment]
> Do you mean that it stops adding new items? That's not what should happen, the expected behavior is that it drops the oldest item in memory to make space for the new ones.

Exactly this. Any idea on how to debug this best?

**Comment by yory8** [comment]
This is very weird. The [code](https://github.com/yory8/clipman/blob/master/storer.go#L9) that deals with it is very simple and straightforward, I don't see how it would fail.

To debug:

1. kill wl-paste, make sure there are no instances running
2. remove clipman's history file
3. set max-items to a low value (eg 3)
4. run the wl-paste command in a terminal (can you share the exact command args you run?) and leave it running aside
5. copy stuff and check if the bug appears again

**Comment by bionade24** [comment]
Ok, this definately works. I also used a 1000 buffer and for-looped `echo $i | wl-copy` into it.  Any other idea why it does get stuck after a long time running the same sway session?

**Comment by yory8** [comment]
It shouldn't get stuck. I never tested on long sessions, and don't know if other users did either, so it's impossible to say what happened. 
Hopefully it was just some corruption in your setup.

Please let me know if it starts happening again. For now, I'm closing this.

**Comment by bionade24** [comment]
This is happening regularly for moths now. When I have time, I'll look into Go debugging and run the debugger permanently to find out.

**Comment by yory8** [comment]
Are you using the latest version of clipman?

I should point out that clipman is not a long running process, but a one-shot program that gets launched afresh whenever there is a copy event. So this bug is very weird.

What I would try is setup a systemd/cron job to kill and restart your wl-paste command at a set time every day, and see if the problem stops happening.

**Comment by bionade24** [comment]
After watching out for this, I actually have times where 1-10 copy buffers don't get stored by clipman and afterwards it's working again. Yes, I have version 1.6.1

**Comment by yory8** [comment]
does it perhaps happen only with a specific application?
aside from this, I have no idea what it might be. Unfortunately I can't reproduce it, so I'm unable to help.

**Comment by bionade24** [comment]
> does it perhaps happen only with a specific application? aside from this, I have no idea what it might be. Unfortunately I can't reproduce it, so I'm unable to help.

No, it definately happens to all programs, from alacritty to chromium.  I really want to get down to this finally, but it'll maybe take a month until I have time spending some time on this.
